client/core/wallet.go

```diff
@@ -177,7 +179,12 @@ func (w *xcWallet) refreshDepositAddress() (string, error) {
 			unbip(w.AssetID))
 	}
 
-	addr, err := w.Address()
+	na, is := w.Wallet.(asset.NewAddresser)
+	if !is {
+		return "", fmt.Errorf("wallet does not generate new addresses")
```

Comment:
Note that `(*Core).NewDepositAddress` will error for eth and any other asset that is not a `NewAddresser`, but the `(*Core).Wallets` method returns `WalletState` with an `Address` field for their current dep addr.

Reply:
Nothing to format. I'm sorry can't resist.